integration/tests/epochs/recover_epoch/suite.go

```diff
@@ -12,10 +24,85 @@ type Suite struct {
 func (s *Suite) SetupTest() {
 	// use a shorter staking auction because we don't have staking operations in this case
 	s.StakingAuctionLen = 2
-	s.DKGPhaseLen = 50
-	s.EpochLen = 250
+	s.DKGPhaseLen = 10
+	s.EpochLen = 80
 	s.EpochCommitSafetyThreshold = 20
 
 	// run the generic setup, which starts up the network
 	s.BaseSuite.SetupTest()
 }
+
+// getNodeInfoDirs returns the internal node private info dir and the node config dir from a container with the specified role.
+func (s *Suite) getNodeInfoDirs(role flow.Role) (string, string) {
+	internalNodePrivInfoDir := fmt.Sprintf("%s/%s", s.GetContainersByRole(role)[0].BootstrapPath(), bootstrap.DirPrivateRoot)
+	nodeConfigJson := fmt.Sprintf("%s/%s", s.GetContainersByRole(role)[0].BootstrapPath(), bootstrap.PathNodeInfosPub)
```

Comment:
⚠️ `GetContainersByRole` is iterating through [`net.Containers`](https://github.com/onflow/flow-go/blob/e1284d8be9d7e2fc6f0170f91fc0063338476f78/integration/testnet/network.go#L144), which is a map. Since maps have random iteration order, we might be accessing different nodes here. 

We can pull out `BootstrapPath` as a variable and call `GetContainersByRole` only once to avoid this problem. We can also make the function in general clearer by documenting this in the `GetContainersByRole` method, or changing the name to reflect that it is returning a randomly ordered list.

Reply:
https://github.com/onflow/flow-go/pull/6156/commits/a46ae41a205ae1d511b720b9bf67130eafd0e01b